src/tests/test_connection.py

```diff
@@ -36,15 +38,134 @@ def test_alias(cleanup):
+    )
+    assert Connection._get_curr_sqlglot_dialect() is None
+
+
+@pytest.mark.parametrize(
+    "sqlalchemy_connection_info, expected_sqlglot_dialect",
+    [
+        (
+            {
+                "dialect": "duckdb",
+                "driver": "duckdb_engine",
+                "server_version_info": [8, 0],
+            },
+            "duckdb",
+        ),
+        (
+            {
+                "dialect": "mysql",
+                "driver": "pymysql",
+                "server_version_info": [10, 10, 3, 10, 3],
+            },
+            "mysql",
+        ),
+        # sqlalchemy and sqlglot have different dialect name, test the mapping dict
+        (
+            {
+                "dialect": "sqlalchemy_mock_dialect_name",
+                "driver": "sqlalchemy_mock_driver_name",
+                "server_version_info": [0],
+            },
+            "sqlglot_mock_dialect",
+        ),
+        (
+            {
+                "dialect": "only_support_in_sqlalchemy_dialect",
+                "driver": "sqlalchemy_mock_driver_name",
+                "server_version_info": [0],
+            },
+            "only_support_in_sqlalchemy_dialect",
+        ),
+    ],
+)
+def test_get_curr_sqlglot_dialect(
+    monkeypatch, sqlalchemy_connection_info, expected_sqlglot_dialect
+):
+    """To test if we can get the dialect name in sqlglot package scope
+
+    Args:
+        monkeypatch (fixture): A convenient fixture for monkey-patching
+        sqlalchemy_connection_info (dict): The metadata about the current dialect
+        expected_sqlglot_dialect (str): Expected sqlglot dialect name
+    """
+    monkeypatch.setattr(
+        Connection,
+        "_get_current_sqlalchemy_connection_info",
+        lambda: sqlalchemy_connection_info,
+    )
+    monkeypatch.setattr(
+        sql.connection,
+        "DIALECT_NAME_SQLALCHEMY_TO_SQLGLOT_MAPPING",
+        {"sqlalchemy_mock_dialect_name": "sqlglot_mock_dialect"},
+    )
+    assert Connection._get_curr_sqlglot_dialect() == expected_sqlglot_dialect
+
+
+@pytest.mark.parametrize(
+    "cur_dialect, expected_support_backtick",
+    [
+        ("mysql", True),
+        ("sqlite", True),
+        ("postgres", False),
+    ],
+)
+def test_is_use_backtick_template(monkeypatch, cur_dialect, expected_support_backtick):
+    """To test if we can get the backtick supportive information from different dialects
+
+    Args:
+        monkeypatch (fixture): A convenient fixture for monkey-patching
+        cur_dialect (bool): Patched dialect name
+        expected_support_backtick (bool): Excepted boolean value to indicate
+        if the dialect supports backtick identifier
+    """
+    monkeypatch.setattr(Connection, "_get_curr_sqlglot_dialect", lambda: cur_dialect)
+    assert Connection.is_use_backtick_template() == expected_support_backtick
+
+
+def test_is_use_backtick_template_sqlglot_missing_dialect(monkeypatch):
```

Comment:
Since `is_use_backtick_template` returns False for every exception, let's add a description/change the test name indicating which exception should be raised (so later, when we'll want to handle it differently - it will be easy to modify).